python/desc/twinkles/sprinkler.py

```diff
@@ -185,255 +184,317 @@ def __init__(self, catsim_cat, visit_mjd, specFileMap, sed_path,
+                default_lensrow[self.defs_dict['galaxyDisk_magNorm']] = 999. #np.nan To be fixed post run1.1
+                default_lensrow[self.defs_dict['galaxyDisk_sedFilename']] = None
+                default_lensrow[self.defs_dict['galaxyBulge_majorAxis']] = 0.0
+                default_lensrow[self.defs_dict['galaxyBulge_minorAxis']] = 0.0
+                default_lensrow[self.defs_dict['galaxyBulge_positionAngle']] = 0.0
+                default_lensrow[self.defs_dict['galaxyBulge_internalAv']] = 0.0
+                default_lensrow[self.defs_dict['galaxyBulge_magNorm']] = 999. #np.nan To be fixed post run1.1
+                default_lensrow[self.defs_dict['galaxyBulge_sedFilename']] = None
+                default_lensrow[self.defs_dict['galaxyBulge_redshift']] = newlens['ZSRC']
+                default_lensrow[self.defs_dict['galaxyDisk_redshift']] = newlens['ZSRC']
+                default_lensrow[self.defs_dict['galaxyAgn_redshift']] = newlens['ZSRC']
+
+            for i in range(newlens['NIMG']):
+                lensrow = default_lensrow.copy()
+                # XIMG and YIMG are in arcseconds
+                # raPhSim and decPhoSim are in radians
+                # Shift all parts of the lensed object,
+                # not just its agn part
+                delta_dec = np.radians(newlens['YIMG'][i] / 3600.0)
+                delta_ra = np.radians(newlens['XIMG'][i] / 3600.0)
+                lens_ra = lensrow[self.defs_dict['raJ2000']]
+                lens_dec = lensrow[self.defs_dict['decJ2000']]
+                lensrow[self.defs_dict['raJ2000']] = lens_ra + delta_ra/np.cos(lens_dec)
+                lensrow[self.defs_dict['decJ2000']] = lens_dec + delta_dec
+                mag_adjust = 2.5*np.log10(np.abs(newlens['MAG'][i]))
+                lensrow[self.defs_dict['galaxyAgn_magNorm']] -= mag_adjust
+                varString = json.loads(lensrow[self.defs_dict['galaxyAgn_varParamStr']])
+                varString[self.defs_dict['pars']]['t0Delay'] = newlens['DELAY'][i]
+                varString[self.defs_dict['varMethodName']] = 'applyAgnTimeDelay'
+                lensrow[self.defs_dict['galaxyAgn_varParamStr']] = json.dumps(varString)
+
+                if self.logging_is_sprinkled:
+                    lensrow[self.defs_dict['galaxyAgn_is_sprinkled']] = 1
+                    lensrow[self.defs_dict['galaxyBulge_is_sprinkled']] = 1
+                    lensrow[self.defs_dict['galaxyDisk_is_sprinkled']] = 1
+
+                #To get back twinklesID in lens catalog from phosim catalog id number
+                #just use np.right_shift(phosimID-28, 10). Take the floor of the last
+                #3 numbers to get twinklesID in the twinkles lens catalog and the remainder is
+                #the image number minus 1.
+                if not isinstance(self.defs_dict['galtileid'], tuple):
+                    lensrow[self.defs_dict['galtileid']] = ((lensrow[self.defs_dict['galtileid']]+int(1.5e10))*100000 +
```

Comment:
This is different than the number scheme I tested because I didn't take into account that we would have > 4 images in some systems. I think this takes us potentially over 2**63. Is this a problem?

Reply:
We also need to include the bit shift correct? But I think we are still ok since it looks like that would get us to `~3*10**18`.

Reply:
good save

but: yes, I think it is still safe as written